compiler/reflect.go

```diff
@@ -126,15 +126,15 @@ type typeCodeAssignmentState struct {
 // that is ever stored in an interface. It tries to use the smallest possible
 // numbers to make the code that works with interfaces as small as possible.
 func (c *Compiler) assignTypeCodes(typeSlice typeInfoSlice) {
-	fn := c.mod.NamedFunction("reflect.ValueOf")
+	/*fn := c.mod.NamedFunction("reflect.ValueOf")
```

Comment:
What this means is that the typecode in interfaces gets slightly larger (as in, a bigger number). This might have minor implications for creating them, but more importantly means that type switches can become slightly bigger because the compiler can't easily create a jump table for them. All this is of course irrelevant for larger programs that are more likely to use reflection.

I did a size comparison using all the tests I could find (5 sets of tests in total), and the size difference was very minor:
* testdata/interface.go got a size increase on wasm (0.8%) and x86 (1.6%), but surprisingly not on a micro:bit test. This is an extreme example so shouldn't be taken too seriously.
* The coroutine and channel tests dropped in size by around 0.6%. I can't explain this, it might be related to the relative immaturity of the wasm backend or to how wasm is very different from other ISAs.
* Four of the driver smoke tests have an increase in size of 16 bytes.
* The compiler smoke tests have no change at all, which is not very surprising as they don't usually use interfaces.

All in all, I would just drop this code instead of cargo-culting it. Perhaps you can leave a note saying there is a possible optimization here that barely helps in real life and causes bugs. Or even better: do this in a separate commit as it is technically separate from Go 1.13 (but is only a real problem in Go 1.13).